Jetstream theme support

fajar-apri-alaska opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #152 #164 

## Summary:

- generate css theme files for jetstream and excursion theme
- css value taken from https://github.com/Alaska-ITS/JetstreamDesignTokens/pull/5 with some modification needed
- added "jetstream" base color
- update "color.base" to "color.jetstream" prefix value in jetstream theme

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

fajar-apri-alaska committed:
feat(theme): support jetstream theme #152

- generate json flat & nested theme files for jetstream theme
- css value taken from
https://github.com/Alaska-ITS/JetstreamDesignTokens/pull/5
with some modification needed
- added "jetstream" base color
- update "color.base" to "color.jetstream" prefix value in jetstream theme

Changes to be committed:
	new file:   scripts/config-jetstream-darkmode.json
	new file:   scripts/config-jetstream.json
	modified:   scripts/styleDictionary.js
	new file:   src/themes/jetstream/color-darkmode/alert.json
	new file:   src/themes/jetstream/color-darkmode/background.json
	new file:   src/themes/jetstream/color-darkmode/base.json
	new file:   src/themes/jetstream/color-darkmode/border.json
	new file:   src/themes/jetstream/color-darkmode/brand.json
	new file:   src/themes/jetstream/color-darkmode/elevation.json
	new file:   src/themes/jetstream/color-darkmode/icon.json
	new file:   src/themes/jetstream/color-darkmode/text.json
	new file:   src/themes/jetstream/color-darkmode/tier.json
	new file:   src/themes/jetstream/color-darkmode/ui.json
	new file:   src/themes/jetstream/color/alert.json
	new file:   src/themes/jetstream/color/background.json
	new file:   src/themes/jetstream/color/base.json
	new file:   src/themes/jetstream/color/border.json
	new file:   src/themes/jetstream/color/brand.json
	new file:   src/themes/jetstream/color/elevation.json
	new file:   src/themes/jetstream/color/icon.json
	new file:   src/themes/jetstream/color/text.json
	new file:   src/themes/jetstream/color/tier.json
	new file:   src/themes/jetstream/color/ui.json
	new file:   src/themes/jetstream/grid-breakpoint.json
	new file:   src/themes/jetstream/text.json

fajar-apri-alaska commented on src/themes/jetstream/color/base.json:
```diff
@@ -0,0 +1,110 @@
+{
+  "color": {
+    "jetstream": {
```
I have pretty much doing the same thing like in [Excursion theme's PR](https://github.com/AlaskaAirlines/AuroDesignTokens/pull/156), but originally, this was `color.base`.
If name changes for this are not needed, then I will update this with all of the references.

braven112 replied:
We just released the first draft of the new excursion theme and made a number of changes in the process, you'll probably need to doublecheck that.  Let us know if you have any trouble mapping css values to the new tier 2 token names.

fajar-apri-alaska reviewed (commented):
All value changes / addition in this PR also needs testing/validation from GD.
Utilizing https://jetstream-rouge.vercel.app/ with hardcoded build files is a way for testing.

fajar-apri-alaska committed:
chore: add comments & hash prefix to color values

Changes to be committed:
	modified:   scripts/config-jetstream-darkmode.json
	modified:   scripts/config-jetstream.json
	modified:   src/themes/jetstream/color/base.json
	modified:   src/themes/jetstream/color/brand.json
	modified:   src/themes/jetstream/color/text.json
	modified:   src/themes/jetstream/color/tier.json
	new file:   src/themes/jetstream/comments.json

fajar-apri-alaska committed:
chore: add configuration css for jetstream #164

- add configuration css for jetstream
- add configuration css for jetstream darkmode

Changes to be committed:
	modified:   scripts/config-jetstream-darkmode.json
	modified:   scripts/config-jetstream.json

braven112 commented:
@fajar-apri-alaska For #164 we'll also need the same CSS file configurations for the Excursion theme so it also has CSS output like Jetstream.

fajar-apri-alaska commented:
> @fajar-apri-alaska For #164 we'll also need the same CSS file configurations for the Excursion theme so it also has CSS output like Jetstream.

I see, then I will provide the same for Excursion theme.

fajar-apri-alaska committed:
chore: update excursion theme config

- Add configuration scripts for excursion theme
to generate css & scss files.

Changes to be committed:
	modified:   scripts/config-excursion.json

jason-capsule42 commented on scripts/config-excursion.json:
```diff
@@ -23,6 +23,194 @@
+      ]
+    },
+
+    "CSSCustomProperties": {
```
This is the only "new" platform that needs to be exported for Excursion. Remove all the others. That should leave the two pre-existing JSON platforms and the CSSCustomProperties platform.

fajar-apri-alaska replied:
All right, only the json and css.

jason-capsule42 commented on scripts/config-jetstream-darkmode.json:
```diff
@@ -0,0 +1,220 @@
+    "./src/themes/jetstream/color/brand.json",
+    "./src/themes/jetstream/comments.json"
+  ],
+  "platforms": {
```
For now we should only be exporting the CSSCustomProperties platform for jetstream (including darkmode)

fajar-apri-alaska replied:
Sure.

jason-capsule42 replied:
@fajar-apri-alaska The JSONVariables are not needed as part of the JetStream theme. That format is part of the Excursion theme but not the JetStream theme.

jason-capsule42 commented on src/themes/jetstream/color-darkmode/alert.json:
```diff
@@ -0,0 +1,56 @@
+{
```
It appears that the Tier2 tokens here were based on the old version of the AuroDesignTokens. While jetstream may have unique brand and base color tokens, the rest of the tokens "must" match the tokens that are currently released in the repo. I can tell that they currently do not because --ds-color-alert... tokens were all deprecated in a recent release.

Please let me know if you need further clarification on what this means for the requested change.

fajar-apri-alaska replied:
I see, thanks for letting me know, I will ask some help from the Jetstream's GD to adjust this.
Will come back to talk to you later.

jason-capsule42 replied:
I think I wasn't clear here @fajar-apri-alaska *all* of these files with Tier 2 tokens (everything except brand.json and base.json) appear to be outdated, not just the alert file.

fajar-apri-alaska replied:
Actually, I got your point on the first time, but I couldn't reach Andri (yet) as it seems He's on leave.
So I tried to double check the json again myself, with the document that He give. In the end, maybe He is the one can confirm if the value is still okay or some value needs to be updated. I will try to reach Him again today.

fajar-apri-alaska committed:
chore: remove unused platform from config file

- Remove unused platform from jetstream and excursion
theme config file, leaving only css platform for jetstream,
and also with json format for excursion.

Changes to be committed:
	modified:   scripts/config-excursion.json
	modified:   scripts/config-jetstream-darkmode.json
	modified:   scripts/config-jetstream.json

fajar-apri-alaska committed:
chore: update tier two tokens from jetstream theme

- update config file for jetstream theme
- remove unused tokens
- update some tokens value per document

Changes to be committed:
	modified:   scripts/config-jetstream-darkmode.json
	modified:   scripts/config-jetstream.json
	renamed:    src/themes/excursion/depth copy.json -> src/themes/excursion/depth.json
	modified:   src/themes/jetstream/color-darkmode/alert.json
	modified:   src/themes/jetstream/color-darkmode/background.json
	modified:   src/themes/jetstream/color-darkmode/base.json
	modified:   src/themes/jetstream/color-darkmode/border.json
	modified:   src/themes/jetstream/color-darkmode/brand.json
	modified:   src/themes/jetstream/color-darkmode/icon.json
	modified:   src/themes/jetstream/color-darkmode/text.json
	modified:   src/themes/jetstream/color-darkmode/ui.json
	modified:   src/themes/jetstream/color/alert.json
	modified:   src/themes/jetstream/color/background.json
	modified:   src/themes/jetstream/color/base.json
	modified:   src/themes/jetstream/color/brand.json
	modified:   src/themes/jetstream/color/text.json
	modified:   src/themes/jetstream/color/tier.json
	modified:   src/themes/jetstream/color/ui.json
	modified:   src/themes/jetstream/text.json

fajar-apri-alaska commented:
Moving this to DRAFT for now, until values are updated and validated.

jason-capsule42 committed:
ci(spellcheck): add spellcheck config file

jason-capsule42 committed:
refactor(jetstream): change  to
